src/components/user_account_menu.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import { Collapse } from 'react-collapse';
+
+// components
+import Button from './button';
+import ProfileImage from './profile_image';
+
+// utils
+import { I18n } from '../utils/i18n';
+
+
+export interface UserAccountMenuProps {
+  /** Called when the sign out button is clicked */
+  onSignOutClickHandler: () => void;
+  /** URL to the account management page */
+  manageAccountURL: string;
+  /** The display name of the user. */
+  displayName: string
+  /** An URL for the user profile image. */
+  imageURL?: string
+}
+
+export interface UserAccountMenuState {
+  showMenu: boolean;
+}
+
+export class UserAccountMenu extends React.Component<UserAccountMenuProps, UserAccountMenuState> {
+
+  private menuRef;
+
+  state = {
+    showMenu: false
+  };
+
+  constructor(props: UserAccountMenuProps) {
+    super(props);
+    this.menuRef = React.createRef();
+    this._onOutsideClickHandler = this._onOutsideClickHandler.bind(this);
+  }
+
+  _setShowMenu(showMenu: boolean): void {
+    if (showMenu) {
+      document.addEventListener('mousedown', this._onOutsideClickHandler);
+    }
+    else {
+      document.removeEventListener('mousedown', this._onOutsideClickHandler);
+    }
+
+    this.setState({
+      showMenu: showMenu
+    });
+  }
+
+  _onOutsideClickHandler() {
+    if (this.menuRef.current && !this.menuRef.current.contains(event.target)) {
+      this._setShowMenu(!this.state.showMenu);
+    }
+  }
+
+  _stopPropagation(e: React.MouseEvent<HTMLDivElement>): void {
+    e.stopPropagation();
+  }
+
+  render(): JSX.Element {
+
+    const { onSignOutClickHandler, manageAccountURL, displayName, imageURL, children } = this.props;
+    const { showMenu } = this.state;
+
+    const classes = classNames('user-account-menu', { 'visible': showMenu });
+
+    return <div className={classes} tabIndex={1} onClick={this._stopPropagation} ref={this.menuRef}>
+
+      <div className="trigger" onClick={() => this._setShowMenu(!showMenu)}>
+        <ProfileImage imageURL={imageURL} size="3.6rem" />
+      </div>
+
+      <div className='container'>
+        <div className="menu">
+          <Collapse isOpened={showMenu} springConfig={{ stiffness: 390, damping: 32 }}>
+
+            <div className="account-section">
+              <div className="account-section__title">{I18n.translate("User account menu title", "My account")}</div>
```

Comment:
We have started implementing the pattern of defining the translations in the top of the file in a `const T`. Should we do this here too?

Reply:
again, should be a future PR to align them